chromedriver_autoinstaller/utils.py

```diff
@@ -163,12 +163,16 @@ def get_chrome_version():
     elif platform == "win":
         # check both of Program Files and Program Files (x86).
         # if the version isn't found on both of them, version is an empty string.
-        dirs = [f.name for f in os.scandir("C:\\Program Files\\Google\\Chrome\\Application") if f.is_dir() and re.match("^[0-9.]+$", f.name)]
-        if dirs:
+        PROGRAMFILES = f"{os.environ['PROGRAMFILES']}\\Google\\Chrome\\Application"
+        PROGRAMFILESX86 = f"{os.environ['PROGRAMFILES(X86)']}\\Google\\Chrome\\Application"
+        
+        path = PROGRAMFILES if os.path.exists(PROGRAMFILES) else PROGRAMFILESX86 if os.path.exists(PROGRAMFILESX86) else None
```

Comment:
If I understand the link in the previous comment correctly I believe `os.environ["ProgramW6432"]` will give us the correct directory for both 32 bit and 64 bit systems.

Maybe we can simplify this code with the following?

```suggestion
        # Define expected path for chrome binary
        path = f"{os.environ['ProgramW6432']}\\Google\\Chrome\\Application"
```

Do you have access to 32bit windows to verify? I only have 64bit, and this change works for me.

Reply:
## 32-bit Windows

On 32-bit windows `os.environ['PROGRAMW6432']`  and `os.environ['PROGRAMFILES(X86)']` returns `None`. As `PROGRAMFILES(X86)` and `PROGRAMW6432` is not defined in 32-bit windows.

Only `os.environ['PROGRAMFILES']` works in 32-bit systems, which returns `C:\Program Files`. 
And `C:\Program Files` is the only program files folder in 32-bit windows

So, For 32-bit windows, Google Chrome Browser will be installed in `C:\Program Files`. Because we can only install 32-bit applications in 32-bit windows

## 64-bit Windows

On 64-bit windows `os.environ['PROGRAMFILES']` returns `C:\Program Files`, which is the default program files folder for any 64-bit application (BUT, in my laptop Google Chrome was not installed here).

And `os.environ['PROGRAMFILES(X86)']` returns `C:\Program Files (x86)` which is dedicated to 32-bit applications in 64-bit windows (In my laptop, Google Chrome was installed here and the reason to raise the issue #71 

## usage

if we use either `os.environ['PROGRAMFILES']` or `os.environ['PROGRAMFILES(X86)']` we can find Google Chrome Installation wherever it is.

but if we use only `os.environ['PROGRAMW6432']` it only checks in 'C:\Program Files' in 64-bit systems, we can't find Google Chrome Installation if it is installed in `C:\Program FIles (x86)`  and 32-bit does not support.


### More:

StackOverflow: [Get path of "Program files" folder that contains 32-bit programs](https://stackoverflow.com/questions/8701967/get-path-of-program-files-folder-that-contains-32-bit-programs) (Read All Answers)

Microsoft: [WOW64](https://learn.microsoft.com/en-us/windows/win32/winprog64/wow64-implementation-details#environment-variables)

Microsoft: [Recognized Environment Variables](https://learn.microsoft.com/en-us/previous-versions/windows/it-pro/windows-vista/cc749104(v=ws.10)?redirectedfrom=MSDN)

StackExchange: [%ProgramFiles% differences on 64bit Windows](https://serverfault.com/questions/413320/programfiles-differences-on-64bit-windows)

Reply:
actually `os.environ['PROGRAMFILES(X86)']` may cause error in 32-bit systems if it is not defined.

I would like to change the code from `os.environ['PROGRAMFILES']` to `os.environ.get("PROGRAMFILES")` and `os.environ['PROGRAMFILES(X86)']` to `os.environ.get("PROGRAMFILES(X86)")`

this will return empty string if variable is not defined.

Reply:
@shaiksamad Great research.

I guess it can't be simple then. LOL.

My concern with os.environ['PROGRAMFILES'] seems to return different results based on the application running it.

From what I'm reading in a variety of our links On 64-bit windows, using 64-bit python os.environ['PROGRAMFILES'] returns C:\Program Files. But on 64-bit windows, using 32-bit python it would return C:\Program Files (x86). 

So in the case of someone trying to get the 64bit chrome version using 32 bit python, I don't think they would get the 'C:\Program Files' path to check if using os.environ['PROGRAMFILES'].

Reply:
Maybe we check if `os.environ['PROGRAMW6432']` exits.

If it does exist, then we search both `os.environ['PROGRAMW6432']` and `os.environ['PROGRAMFILES(X86)']` for chrome.

Else, we just search os.environ['PROGRAMFILES'] for chrome.

Reply:
I tested this on 64bit windows with both 32bit chrome and 64bit chrome. What do you think?

```

        if 'ProgramW6432' in os.environ:
            # On 64bit windows.
            program_files_options = ['ProgramW6432', 'ProgramFiles(x86)']
        else:
            # on 32bit windows
            program_files_options = ['ProgramFiles']

        version = None
        for option in program_files_options:
            # Check if check if chrome application folder exists for the given program files folder
            chrome_app_path = f"{os.environ[option]}\\Google\\Chrome\\Application"
            if os.path.exists(chrome_app_path):
                # Find folders specifying version number in the chrome app folder
                dirs = [f.name for f in os.scandir(chrome_app_path) if f.is_dir() and re.match("^[0-9.]+$", f.name)]
                version = max(dirs) if dirs else None
                if version:
                    break
```

Reply:
we can update code to:

```python
PROGRAMFILES =  f"{os.environ.get('ProgramW6432') or os.environ.get('ProgramFiles')}\\Google\\Chrome\\Application"
PROGRAMFILESX86 = f"{os.environ.get('PROGRAMFILES(X86)')}\\Google\\Chrome\\Application"
        
path = PROGRAMFILES if os.path.exists(PROGRAMFILES) else PROGRAMFILESX86 if os.path.exists(PROGRAMFILESX86) else None

```

this works fine on 32-bit windows, 64-bit windows with 32 and 64 bit python

Reply:
Beautiful solution @shaiksamad! Only thing I'd suggest is some comments below. Up to you if you want to add comments.

Add your update to the PR and I'll merge it.

```
# Define standard 'Program Files' path for 64bit OS or 32Bit OS
PROGRAMFILES =  f"{os.environ.get('ProgramW6432') or os.environ.get('ProgramFiles')}\\Google\\Chrome\\Application"

# Define 32bit 'Program Files' path for 64bit OS
PROGRAMFILESX86 = f"{os.environ.get('PROGRAMFILES(X86)')}\\Google\\Chrome\\Application"
        
path = PROGRAMFILES if os.path.exists(PROGRAMFILES) else PROGRAMFILESX86 if os.path.exists(PROGRAMFILESX86) else None
```